### PR TITLE
Feature/allisonbhavsar/52 calendar feed api

### PR DESCRIPTION
**Pull request by allisonbhavsar**
### Administrative Info

Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Github Issue]-[3-4 word description separated by dashes]`.

### What issue(s) does this branch close?
Issue #52 

### Changes

What changes did you make?

- Added /api/event-feed/ route to get calendar event feed for a particular user

### Testing

How did you confirm your changes worked?

- Tested with and without specifying userId param
- Tested with invalid date range (start > end)
- Tested with nonexistent userId
- Tested error cases with missing params

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

`npm run test`

### CODE REVIEW

File: __test__/event-feed.test.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  await client.query(
+    "INSERT INTO users(id, email, role, first_name, last_name, phone_number, address) VALUES('user_a', 'emaila@gmail.com', 'role', 'fname', 'lname', '#', 'addr')"
+  );
+  await client.query(
+    "INSERT INTO commitments(user_id, event_information_id) VALUES('user_a', 'id_a')"
+  );
+});
+
+afterAll(async () => {
+  await client.end();
```
Comment (Anshul-Birla) [inline __test__/event-feed.test.ts:50]: lets delete all the data from tables that you accessed so the next tests are guaranteed to run with a clean db. This wouldn't be an issue if jest ran tests in the same order, but sometimes it doesn't :(

Nishant is currently making a script that will enter a ton of dummy data into the db for our testing purposes :)

File: __test__/event-feed.test.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+});
+
+describe("[GET] /api/event-feed", () => {
+  test("get calendar event feed for a user", async () => {
+    const body = {
+      start: "2022-02-25 21:11:45-08",
+      end: "2022-02-28 21:11:45-08",
+      userId: "user_a",
+    };
```
Comment (Anshul-Birla) [inline __test__/event-feed.test.ts:59]: Oof, im sorry if i messed up giving this information out. The API takes in nothing in the body. Everything is done via query parameters. So everything is the same, except these should be passed in as the query parameter of  the http request

File: pages/api/event-feed/index.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+import { getEventFeed } from "../../../lib/database/calendar";
+import { StatusCodes } from "http-status-codes";
+
+// handles requests to /api/event-feed/
+const eventFeedHandler: NextApiHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  switch (req.method) {
```
Comment (Anshul-Birla) [inline pages/api/event-feed/index.ts:7]: here, `start`, `end` and `userId` will be specified within the query parameters of the request, not the body. Something like this should do the trick:

``` javascript
if (!req.query) {
    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json("Internal Server Error");
  }

  
  const start = req.query.start as string;
  const end = req.query.end as string;
  const userId = req.query.userId as string;

  if (!start || !end) {
    return res.status(400).json("No start of end date specified");
  }
```

**Review by Anshul-Birla — CHANGES_REQUESTED** [review]
Super awesome work getting this all sorted out very quickly. A change that I pointed out - the API is taking the start, end, and userID dates in the query parameters of the call, not the body. I pointed out the changes you need to make for it to all work out :)

**Review by Anshul-Birla — COMMENTED** [review]
Just ignore this I forgot to mark the files as viewed lol

File: lib/database/calendar.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import { client } from "../db";
```
Comment (Anshul-Birla) [inline lib/database/calendar.ts:1]: Forgot about this. can we put this in calendar-events.ts? That way the file the database functions are in correspond to the route they belong to.

**Review by Anshul-Birla — CHANGES_REQUESTED** [review]

**Review by Anshul-Birla — APPROVED** [review]
I love it! Great work :)

**Review by ryanbuui — APPROVED** [review]
LGTM Nice work!

**Pull request merged**